demo/src/test/java/com/example/demo/controller/ShopItemTest.java

```diff
@@ -0,0 +1,106 @@
+package com.example.demo.controller;
+
+import com.example.demo.model.ShopItem;
+import com.example.demo.repository.WebshopRepository;
+import com.example.demo.service.WebshopService;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.Date;
+import java.util.List;
+
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+class ShopItemTest {
+    @InjectMocks
+    private WebshopService webshopService;
+
+    @Mock
+    private WebshopRepository webshopRepository;
+
+    @Test
+    void shouldFindByPrice() {
+        //given
+        final var name1 = "banana";
+        final var priceOfItem = 10;
+        final Date date = null;
+
+        //when
+        when(webshopRepository.findByPrice(priceOfItem)).thenReturn(
+                List.of(ShopItem.builder()
+                        .id(1)
+                        .name(name1)
+                        .price(priceOfItem)
+                        .quantity(1)
+                        .date(null)
+                        .build()));
+        final var shopItems = webshopRepository.findByPrice(priceOfItem);
+
+        //then
+        Assertions.assertEquals(1, shopItems.size());
+        Assertions.assertEquals(name1, shopItems.get(0).getName());
+        Assertions.assertEquals(priceOfItem, shopItems.get(0).getPrice());
+    }
+
+    @Test
+    void shouldFindByPriceAndQuantity() {
+        //given
+        final var quantityOfItem = 5;
+        final var priceOfItem = 10;
+        final Date date = null;
+
+        //when
+        when(webshopRepository.findByPriceAndQuantity(priceOfItem, quantityOfItem)).thenReturn(
+                List.of(ShopItem.builder()
+                        .id(1)
+                        .name("banana")
+                        .price(priceOfItem)
+                        .quantity(quantityOfItem)
+                        .date(null)
+                        .build()));
+        final var shopItems = webshopRepository.findByPriceAndQuantity(priceOfItem, quantityOfItem);
```

Comment:
This does not make any sense. You are mocking the repo and after asserting for the same thing - please fix your tests. Your test name is also odd and you are not testing the controller, so don't put it in the controller layer :)

Reply:
You have replaced and renamed your test - that's good; but you are still not testing your service! Remember: you want to test the service individually. Then you have to mock everything that the service depends on (in this case the repo) and test the service in an isolated way.

Now you are mocking the repo and subsequently testing the repo - which doesn't make any sense.